### ISSUE TITLE
Table column related WCS

### ISSUE THREAD

**Issue by olebole** [astropy/astropy#14884]
### What is the problem this feature will solve?

I have a number of (FITS) tables that use WCS to describe arrays within table cells, as described in the [WCS paper](https://arxiv.org/pdf/astro-ph/0207407.pdf) section 3.1. Here basically, each column is associated with a WCS header that has a format slightly differing from the "normal" (global) one. In my case, this is like
```
1CTYP1  = 'AWAV    '           / Grating dispersion function (air)
1CUNI1  = 'Angstrom'           / Wavelength unit
1CRPX1  =                    1 / Pixel zeropoint
1CRVL1  =               3700.0 / Wavelength zeropoint
1CDLT1  =                 0.25 / Wavelength increment
```
where the leftmost "1" is the axis, and the rightmost one is the column index. See table 2 of the paper for a full list of keywords here.

I would like to be able to handle these WCS header within Astropy.

Asking on Slack showed that this is probably not implemented yet.

### Describe the desired outcome

The optimal outcome would be if this could be transparently handled within a table (column). 

Probably the optimal variant would be if the WCS related headers would be bound to the column and renamed into their "usual" variants when a file was read in (i.e. "1CTYP2" --> "CTYPE1"). When writing a FITS file, the column-specific names could be re-generated. This would ensure that the WCS keywords stay correct even when the columns were re-arranged. Also, this would expose familiar keywords to the user, and to the WCS subpackage. Also, exporting table cells as images with a proper WCS would be simple.

The WCS subpackage then could take them as 
```Python
from astropy.wcs import WCS

flux_wcs = WCS(tbl['flux'].meta)
```
or similar without larger changes there.

**Comment by pllim** [comment]
tl;dr -- I cannot find "1CTYP1" in that PDF. Is this naming convention something the instrument made up, or part of some official standard?

Also, looks like a spectrum, so are you sure this isn't better suited as some custom `specutils` reader downstream?

**Comment by hamogu** [comment]
This is probably related to https://github.com/astropy/astropy/issues/8389 and https://github.com/astropy/astropy/issues/11413.
